metadata_test.go

```diff
@@ -0,0 +1,500 @@
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	it.After(func() {
+		Expect(os.RemoveAll(tmpDir)).To(Succeed())
+	})
+
+	it("builds full stack", func() {
+		var buildReleaseDate, runReleaseDate time.Time
+
+		by("confirming that the build image is correct", func() {
+			dir := filepath.Join(tmpDir, "build-index")
+			err := os.Mkdir(dir, os.ModePerm)
+			Expect(err).NotTo(HaveOccurred())
+
+			archive, err := os.Open(stack.BuildArchive)
+			Expect(err).NotTo(HaveOccurred())
+			defer archive.Close()
+
+			err = vacation.NewArchive(archive).Decompress(dir)
+			Expect(err).NotTo(HaveOccurred())
+
+			path, err := layout.FromPath(dir)
+			Expect(err).NotTo(HaveOccurred())
+
+			index, err := path.ImageIndex()
+			Expect(err).NotTo(HaveOccurred())
+
+			indexManifest, err := index.IndexManifest()
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(indexManifest.Manifests).To(HaveLen(1))
+			Expect(indexManifest.Manifests[0].Platform).To(Equal(&v1.Platform{
+				OS:           "linux",
+				Architecture: "amd64",
+			}))
+
+			image, err := index.Image(indexManifest.Manifests[0].Digest)
+			Expect(err).NotTo(HaveOccurred())
+
+			file, err := image.ConfigFile()
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(file.Config.Labels).To(SatisfyAll(
+				HaveKeyWithValue("io.buildpacks.stack.id", "io.buildpacks.stacks.bionic"),
+				HaveKeyWithValue("io.buildpacks.stack.description", "ubuntu:bionic with many common C libraries and utilities"),
+				HaveKeyWithValue("io.buildpacks.stack.distro.name", "ubuntu"),
+				HaveKeyWithValue("io.buildpacks.stack.distro.version", "18.04"),
+				HaveKeyWithValue("io.buildpacks.stack.homepage", "https://github.com/paketo-buildpacks/bionic-full-stack"),
+				HaveKeyWithValue("io.buildpacks.stack.maintainer", "Paketo Buildpacks"),
+				HaveKeyWithValue("io.buildpacks.stack.metadata", MatchJSON("{}")),
+				HaveKeyWithValue("io.buildpacks.stack.mixins", ContainSubstring(`"ca-certificates"`)),
+				HaveKeyWithValue("io.paketo.stack.packages", ContainSubstring(`"jq"`)),
+			))
+
+			buildReleaseDate, err = time.Parse(time.RFC3339, file.Config.Labels["io.buildpacks.stack.released"])
+			Expect(err).NotTo(HaveOccurred())
+			Expect(buildReleaseDate).NotTo(BeZero())
+
+			Expect(image).To(SatisfyAll(
+				HaveFileWithContent("/etc/group", ContainSubstring("cnb:x:1000:")),
+				HaveFileWithContent("/etc/passwd", ContainSubstring("cnb:x:1000:1000::/home/cnb:/bin/bash")),
+				HaveDirectory("/home/cnb"),
+			))
+
+			Expect(file.Config.User).To(Equal("1000:1000"))
+
+			Expect(file.Config.Env).To(ContainElements(
+				"CNB_USER_ID=1000",
+				"CNB_GROUP_ID=1000",
+				"CNB_STACK_ID=io.buildpacks.stacks.bionic",
+			))
+
+			Expect(image).To(HaveFileWithContent("/etc/gitconfig", ContainLines(
+				"[safe]",
+				"\tdirectory = /workspace",
+				"\tdirectory = /workspace/source-ws",
+				"\tdirectory = /workspace/source",
+			)))
+
+			Expect(image).To(HaveFileWithContent("/var/lib/dpkg/status", SatisfyAll(
```

Comment:
This seems like a reasonable test to me. @robdimsdale are you aligned with this strategy of checking all the packages in the Jammy Full builder as well?

Reply:
Yup, this is what we're doing over in the Jammy Full stack too.

I am not 100% convinced of the value of this, as it feels like we're essentially writing a test to parse a list (rather than validating specific behavior), but i recognize that it provides us with a fast feedback loop if we remove a package.

So, on balance, I'm happy with this type of test.

Reply:
@robdimsdale agreed